CSC crashes (instead of emitting error CS0154) when dotting off a setter-only property of type dynamic

**Version Used**: 
VS2015 Update 2 RC

**Steps to Reproduce**:

Create a console app with the following code, and build it:

```cs
class Program
{
    static void Main()
    {
        I i = null;
        System.Type ts = i.s.GetType();  // compile-time error CS0154
        System.Type td = i.d.GetType();  // compile-time CRASH with error code -2146232797
    }

    public interface I
    {
        string s { set; }
        dynamic d { set; }
    }
}
```

**Expected Behavior**:

Both `GetType()` lines should emit CS0154 "property cannot be used in this context because it lacks a get accessor"

**Actual Behavior**:

The second `GetType()` line causes a compiler crash with

--- a/src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs
+++ b/src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs
@@ -4510,6 +4510,12 @@ namespace Microsoft.CodeAnalysis.CSharp
 
             if ((object)leftType != null && leftType.IsDynamic())
             {
+                // There are some sources of a `dynamic` typed value that can be known before runtime
+                // to be invalid. For example, accessing a set-only property whose type is dynamic:
+                //   dynamic Foo { set; }
+                // If Foo itself is a dynamic thing (e.g. in `x.Foo.Bar`, `x` is dynamic, and we're
+                // currently checking Bar), then CheckValue will do nothing.
+                boundLeft = CheckValue(boundLeft, BindValueKind.RValue, diagnostics);
                 return BindDynamicMemberAccess(node, boundLeft, right, invoked, indexed, diagnostics);
             }
 

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/DynamicTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/DynamicTests.cs
@@ -3434,6 +3434,33 @@ class Test
             CompileAndVerify(compilation2, expectedOutput: @"4");
         }
 
+        [Fact, WorkItem(9945, "https://github.com/dotnet/roslyn/issues/9945")]
+        public void DynamicGetOnlyProperty()
+        {
+            string source = @"
+class Program
+{
+    static void Main()
+    {
+        I i = null;
+        System.Type t = i.d.GetType();
+    }
+
+    interface I
+    {
+        dynamic d { set; }
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib(source, new[] { CSharpRef, SystemCoreRef }, options: TestOptions.DebugDll);
+            // crash happens during emit if not detected, so VerifyDiagnostics (no Emit) doesn't catch the crash.
+            compilation.VerifyEmitDiagnostics(
+                // (7,25): error CS0154: The property or indexer 'Program.I.d' cannot be used in this context because it lacks the get accessor
+                //         System.Type t = i.d.GetType();
+                Diagnostic(ErrorCode.ERR_PropertyLacksGet, "i.d").WithArguments("Program.I.d").WithLocation(7, 25)
+            );
+        }
+
         [ClrOnlyFact(ClrOnlyReason.Ilasm)]
         public void IncorrectArrayLength()
         {

